src/frontend/src/optimizer/plan_node/stream_materialize.rs

```diff
@@ -86,9 +89,11 @@ impl StreamMaterialize {
             Distribution::Single => RequiredDist::single(),
             _ => {
                 if is_index {
-                    RequiredDist::PhysicalDist(Distribution::HashShard(
-                        user_order_by.field_order.iter().map(|x| x.index).collect(),
-                    ))
+                    assert_matches!(
+                        user_distributed_by,
```

Comment:
Is this field applicable if we're to create a plain materialized view? If not, how about make it an `Option` to imply `is_index`?

Reply:
It is only used by create index currently, but I think it is also applicable to plain materialized view in the future. How about Adding an assert for plain materialized view? (Assert RequiredDist == Any).

Reply:
It seems all of PlanRoot is created with `required_dist = Any`. I will test it.